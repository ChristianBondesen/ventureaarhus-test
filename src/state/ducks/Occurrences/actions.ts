import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

// Interface for your actions -->
export interface ISetOccurrences extends IFluxStandardAction {
  type: types.SET_OCCURRENCES;
  payload: types.IOccurrence[];
}

export interface ISetOccurrencesTags extends IFluxStandardAction {
  type: types.SET_OCCURRENCES_TAGS;
  payload: types.ITags[];
}

export interface IUpdateRecommendedOccurrences extends IFluxStandardAction {
  type: types.UPDATE_RECOMMENDED_OCCURRENCES;
  payload: types.IOccurrence[];
}

// State of your Actions -->
// (Import for the reducer)
export type OccurrenceStateAction =
  | ISetOccurrences
  | ISetOccurrencesTags
  | IUpdateRecommendedOccurrences;

// Actual action implementing the former defined interface -->
export const setOccurrenceList = (
  occurrences: types.IOccurrence[]
): ISetOccurrences => ({
  type: types.SET_OCCURRENCES,
  payload: occurrences,
});

export const setOccurrencesTags = (
  tags: types.ITags[]
): ISetOccurrencesTags => ({
  type: types.SET_OCCURRENCES_TAGS,
  payload: tags,
});

export const updateRecommendedOccurrences = (
  recOccurs: types.IOccurrence[]
): IUpdateRecommendedOccurrences => ({
  type: types.UPDATE_RECOMMENDED_OCCURRENCES,
  payload: recOccurs,
});

// Interface of all your Actions -->
export interface IOccurrenceActions {
  setOccurrenceList(occurrences: types.IOccurrence[]);
  setOccurrencesTags(tags: types.ITags[]);
  updateRecommendedOccurrences(recOccurs: types.IOccurrence[]);
}

// Export the actions -->
export default {
  setOccurrenceList,
  setOccurrencesTags,
  updateRecommendedOccurrences,
};
