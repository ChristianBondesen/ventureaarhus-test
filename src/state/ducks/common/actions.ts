import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

export interface IChangeConnectionStatus extends IFluxStandardAction {
  type: types.CHANGE_CONNECTION_STATUS;
  payload: boolean;
}
export interface IIncrementNetworkCalls extends IFluxStandardAction {
  type: types.INCREMENT_NETWORK_CALLS;
}

export type CommonStateAction =
  | IChangeConnectionStatus
  | IIncrementNetworkCalls;

export const setConnectionStatus = (
  connected: boolean
): IChangeConnectionStatus => ({
  type: types.CHANGE_CONNECTION_STATUS,
  payload: connected,
});

export const incrementNetworkCalls = (): IIncrementNetworkCalls => ({
  type: types.INCREMENT_NETWORK_CALLS,
});

export interface ICommonActions {
  setConnectionStatus(connected: boolean);
  incrementNetworkCalls();
}

export default {
  setConnectionStatus,
  incrementNetworkCalls,
};
