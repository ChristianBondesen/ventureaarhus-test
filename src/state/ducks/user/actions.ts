import { IFluxStandardAction } from '../FluxStandardAction';
import * as types from './types';

export interface ISetUser extends IFluxStandardAction {
  type: types.SET_USER;
  payload: types.IUser;
}

export type UserStateAction = ISetUser;

export const setUser = (user: types.IUser): ISetUser => ({
  type: types.SET_USER,
  payload: user,
});

export interface IUserActions {
  setUser(user: types.IUser);
}

export default {
  setUser,
};
