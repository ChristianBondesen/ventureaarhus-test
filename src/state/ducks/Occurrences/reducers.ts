import { OccurrenceStateAction } from './actions';
import * as types from './types';

// Interface for defualt state
export interface IOccurrenceState {
  occurrencesList: types.IOccurrence[];
  tags: types.ITags[];
}

// Default state
export const defaultState: IOccurrenceState = {
  occurrencesList: [],
  tags: [],
};

// The reducer
export const occurrencesReducer = (
  state: IOccurrenceState = defaultState,
  action: OccurrenceStateAction
): IOccurrenceState => {
  switch (action.type) {
    case types.SET_OCCURRENCES: {
      return {
        ...state,
        occurrencesList: action.payload,
      };
    }
    case types.SET_OCCURRENCES_TAGS: {
      return {
        ...state,
        tags: action.payload,
      };
    }

    default:
      return state;
  }
};

export default occurrencesReducer;
