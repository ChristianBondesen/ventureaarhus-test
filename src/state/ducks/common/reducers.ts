import { CommonStateAction } from './actions';
import * as types from './types';

export interface ICommonState {
  isConnected: boolean;
  networkCalls: number;
}

const TAG = 'reducers.ts';

export const defaultState: ICommonState = {
  isConnected: true,
  networkCalls: 0,
};

export const commonReducer = (
  state: ICommonState = defaultState,
  action: CommonStateAction
): ICommonState => {
  switch (action.type) {
    case types.CHANGE_CONNECTION_STATUS:
      return {
        ...state,
        isConnected: action.payload,
      };

    case types.INCREMENT_NETWORK_CALLS:
      return {
        ...state,
        networkCalls: state.networkCalls + 1,
      };

    default:
      return state;
  }
};

export default commonReducer;
