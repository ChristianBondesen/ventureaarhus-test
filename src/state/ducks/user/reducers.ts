import { UserStateAction } from './actions';
import * as types from './types';
import { IEvent } from '../Event/types';

export interface IUserState {
  id: number;
  name: string;
  thumbnail: string;
  likedEvents: IEvent[];
  followedEvents: IEvent[];
  participatingEvents: IEvent[];
  token: string;
}

const TAG = 'reducers.ts';

// Default state of the userstate
export const defaultState: IUserState = {
  id: 0,
  name: '',
  thumbnail: '',
  likedEvents: [],
  followedEvents: [],
  participatingEvents: [],
  token: '',
};

export const userReducer = (
  state: IUserState = defaultState,
  action: UserStateAction
): IUserState => {
  switch (action.type) {
    case types.SET_USER: {
      return {
        ...state,
        id: action.payload.id,
        name: action.payload.name,
        thumbnail: action.payload.thumbnail,
        participatingEvents: action.payload.participatingEvent,
        followedEvents: action.payload.followedEvents,
        likedEvents: action.payload.likedEvents,
      };
    }

    case types.SET_TOKEN: {
      return {
        ...state,
        token: action.payload,
      };
    }

    case types.SET_LIKED_EVENTS: {
      return {
        ...state,
        likedEvents: action.payload,
      };
    }
    default: {
      return state;
    }
  }
};

export default userReducer;
