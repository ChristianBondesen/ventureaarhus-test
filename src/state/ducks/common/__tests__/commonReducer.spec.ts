import * as actions from './../actions';
import { commonReducer, defaultState } from './../reducers';

const TAG = 'commonReducer.spec.ts';

const initialState = defaultState;

describe('Common reducer', () => {
  it('should set connection boolean', () => {
    const action = actions.setConnectionStatus(true);
    expect(commonReducer(undefined, action)).toMatchSnapshot();
  });
});
