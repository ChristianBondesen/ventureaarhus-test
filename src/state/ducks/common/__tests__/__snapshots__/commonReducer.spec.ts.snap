// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common reducer should set connection boolean 1`] = `
Object {
  "isConnected": true,
}
`;
